spec/requests/people_n_plus_one_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "n+1 queries in the people controller" do
+  describe "#index", n_plus_one: true do
+    context "when viewing people in a collection" do
+      let!(:collection) { create(:collection) }
+
+      populate do |n|
+        create_list(:collection_participant, n, collection: collection)
+      end
+
+      subject do
+        proc do
+          get collection_people_path(collection_id: collection)
+        end
+      end
+
+      warmup { subject.call }
+
+      it "produces a constant number of queries" do
+        expect { subject.call }
+          .to perform_constant_number_of_queries
+      end
+    end
+  end
+
+  describe "#search", n_plus_one: true, pseud_search: true do
+    context "when there are search results" do
+      populate do |n|
+        create_list(:pseud, n, name: "nplusone")
+        run_all_indexing_jobs
+      end
+
+      subject do
+        proc do
+          get search_people_path, params: { "people_search" => { "name" => "nplusone" } }
+        end
+      end
+
+      warmup { subject.call }
+
+      # TODO: AO3-6743, ideally tis would be a constant number of queries too.
+      # However, I'm only testing that we didn't add more with ActiveStorage
+      # now, as those changes are already very involved.
+      # - Brian Austin, June 2024
+      it "produces about 1 query per pseud" do
+        expect { subject.call }
+          .to perform_linear_number_of_queries(slope: 1)
```

Comment:
Locally this passes for me even with `perform_constant_number_of_queries`.

Reply:
:thinking: it consistently fails for me... seeing what happens in CI, if it passes there (and maybe in Gitpod?) I'll leave it on and close that story

Reply:
Looks like Github actions also runs into what I'm seeing locally: https://github.com/otwcode/otwarchive/actions/runs/9726260721/job/26844510622

Reply:
Ah sorry, I meant with the `pseud_search: true` removed and the `PseudIndexer.prepare_for_testing` added to exactly mimic what `works_n_plus_one_spec.rb` does. If I don't do those additional changes it also doesn't pass for me.

Reply:
Ah got it, thanks! Looks like that was it!